### DISCUSSION TITLE
QDialog inside a custom component being reused?

### DISCUSSION THREAD

**Discussion by Namidas** [quasarframework/quasar#13566] [General - Components / Directives / etc] [answered]
Hi, I have a custom component with a QDialog inside.
When there's a single instance of said component, everything works perfectly, but when there are more than one instance of that component, then it's reusing the dialog, or somehow reusing it's props.
Something like this...

the component:

```
<div>
        UID: {{uid}}
       <q-dialog>
              <div>{{uid}}</div>
       </q-dialog>
</div>
```

And then something like:

```
<my-compo uid="test_1" />
<my-compo uid="test_2" />
```

But then on the redered component body you see the correct UID value for each instance, but the dialog always shows "test_2" as the UID (and yes, it's slightly more extensive, everything wrapped on a QCard, etc etc etc).

Any thoughts on this?

**Comment by metalsadman** [comment]
Make a codepen

**Comment by Namidas** [comment, marked as answer]
Solved this...
I was using a global constant with default setup data definitions, so I was getting a nice (yet not desired) global reactivity between any and all instances of that component, specifically, the default value I'm talking about was actually the v-model for the dialog, so when I opened one, actually all of them inside all of the instances of my custom component were opened at once, and I kept getting this "test_2" UID error because all the dialogs were one in top of the other, so I was looking at the last one defined.

As said, in this case it was a problem, but looks like a nice way to get that kind of global reactivity, if you ever need it, something like this...

```
const myGlobal = ref('some-value');

myCompo = defineComponent({
     setup (props) {
          return {
                globalReactiveValue: myGlobal
          }
     }
});
```

It's also a way to get some 2 direction reactivity without having to work with events, watchers, models, etc.
